### ISSUE TITLE
Flag Synchronous option in XMLHttpRequest.open

### ISSUE THREAD

**Issue by erosman** [mozilla/addons-linter#2086]
Example:
https://reviewers.addons.mozilla.org/en-US/firefox/files/compare/1002571...991553/file/bgr.js#L12

```
httpGet = function(url)
{
    var xmlHttp = new XMLHttpRequest();
    xmlHttp.open( "GET", url, false );
    xmlHttp.send( null );
    return xmlHttp.responseText;
}
```

**Comment by EnTeQuAk** [comment]
Why should we flag this? Is that a rule in our policy somewhere?

I'm sure there are good reasons to flag this, e.g synchronous calls may block user ui etc but generally, it's hard to make this false-positive proof since the synchronous request can be hidden in some async code somewhere.

Not sure what the benefit would be here.

cc @wagnerand

**Comment by wagnerand** [comment]
@EnTeQuAk I think your example talks about a false negative. I think we can live with that since we'll never reach 100% coverage using those static checks and this warning is to point out obvious mistakes instead of trying to identify sneaky behavior.

P3 is fine though, it's not a high priority.
